src/network/networkd-dhcp-common.c

```diff
@@ -460,10 +460,29 @@ int config_parse_dhcp_send_option(
         }
 
         p = rvalue;
+        if (ltype == AF_INET6 && streq(lvalue, "SendVendorOption")) {
+                r = extract_first_word(&p, &word, ":", 0);
+                if (r == -ENOMEM)
+                        return log_oom();
+                if (r <= 0 || isempty(p)) {
+                        log_syntax(unit, LOG_ERR, filename, line, r,
+                                   "Invalid DHCP option, ignoring assignment: %s", rvalue);
+                        return 0;
+                }
+
+                r = safe_atou32(word, &enterprise_identifier);
+                if (r < 0) {
+                        log_syntax(unit, LOG_ERR, filename, line, r,
+                                   "Failed to parse DHCPv6 enterprise identifier data, ignoring assignment: %s", p);
+                        return 0;
+                }
+                word = mfree(word);
+        }
+
         r = extract_first_word(&p, &word, ":", 0);
         if (r == -ENOMEM)
                 return log_oom();
-        if (r <= 0) {
+        if (r <= 0 || isempty(p)) {
```

Comment:
Same here. Why do you add this?

Reply:
This will crash 
```
[DHCPv6]
SendVendorOption=1234:11:string:hello
SendVendorOption=3456:12:string:hello1
SendVendorOption=11:1:string:
SendVendorOption=12:1:uint8:
SendVendorOption=13:1:uint16:
SendVendorOption=14:1:uint32:
SendVendorOption=15:1:ipv4address:
SendVendorOption=16:1:ipv6address:

gdb) bt
#0  0x00007ffff7be76a5 in __strlen_avx2 () from /lib64/libc.so.6
#1  0x000000000046822d in cunescape (s=0x0, flags=UNESCAPE_ACCEPT_NUL, ret=0x7fffffffc668) at ../src/basic/escape.h:55
#2  0x000000000046a6b6 in config_parse_dhcp_send_option (unit=0x0, filename=0x55d600 "/usr/lib/systemd/network/veth99.network", line=17, section=0x55e9e0 "DHCPv6", section_line=14, 
    lvalue=0x562540 "SendVendorOption", ltype=10, rvalue=0x562551 "11:1:string:", data=0x563658, userdata=0x5634b0) at ../src/network/networkd-dhcp-common.c:597
#3  0x00007ffff7d2fcc7 in next_assignment (unit=0x0, filename=0x55d600 "/usr/lib/systemd/network/veth99.network", line=17, lookup=0x7ffff7d2f76d <config_item_perf_lookup>, 
    table=0x4392f6 <network_network_gperf_lookup>, section=0x55e9e0 "DHCPv6", section_line=14, lvalue=0x562540 "SendVendorOption", rvalue=0x562551 "11:1:string:", 
    flags=CONFIG_PARSE_WARN, userdata=0x5634b0) at ../src/shared/conf-parser.c:132
#4  0x00007ffff7d306f8 in parse_line (unit=0x0, filename=0x55d600 "/usr/lib/systemd/network/veth99.network", line=17, sections=0x4f4a18 "Match", 
    lookup=0x7ffff7d2f76d <config_item_perf_lookup>, table=0x4392f6 <network_network_gperf_lookup>, flags=CONFIG_PARSE_WARN, section=0x7fffffffc998, section_line=0x7fffffffc940, 
    section_ignored=0x7fffffffc93d, l=0x562540 "SendVendorOption", userdata=0x5634b0) at ../src/shared/conf-parser.c:270
#5  0x00007ffff7d30da4 in config_parse (unit=0x0, filename=0x55d600 "/usr/lib/systemd/network/veth99.network", f=0x55d970, sections=0x4f4a18 "Match", 
    lookup=0x7ffff7d2f76d <config_item_perf_lookup>, table=0x4392f6 <network_network_gperf_lookup>, flags=CONFIG_PARSE_WARN, userdata=0x5634b0) at ../src/shared/conf-parser.c:395
#6  0x00007ffff7d3103a in config_parse_many_files (conf_file=0x55d600 "/usr/lib/systemd/network/veth99.network", files=0x563e50, sections=0x4f4a18 "Match", 
    lookup=0x7ffff7d2f76d <config_item_perf_lookup>, table=0x4392f6 <network_network_gperf_lookup>, flags=CONFIG_PARSE_WARN, userdata=0x5634b0) at ../src/shared/conf-parser.c:452
#7  0x00007ffff7d313e5 in config_parse_many (conf_file=0x55d600 "/usr/lib/systemd/network/veth99.network", conf_file_dirs=0x7fffffffcc40, 
    dropin_dirname=0x7fffffffcb70 "veth99.network.d", sections=0x4f4a18 "Match", lookup=0x7ffff7d2f76d <config_item_perf_lookup>, table=0x4392f6 <network_network_gperf_lookup>, 
    flags=CONFIG_PARSE_WARN, userdata=0x5634b0) at ../src/shared/conf-parser.c:511
#8  0x0000000000420bd2 in network_load_one (manager=0x5537e0, networks=0x553850, filename=0x55d600 "/usr/lib/systemd/network/veth99.network") at ../src/network/networkd-network.c:470
#9  0x0000000000421027 in network_load (manager=0x5537e0, networks=0x553850) at ../src/network/networkd-network.c:567
#10 0x0000000000419c46 in manager_load_config (m=0x5537e0) at ../src/network/networkd-manager.c:1938
#11 0x000000000040d78f in run (argc=1, argv=0x7fffffffd948) at ../src/network/networkd.c:87
#12 0x000000000040de46 in main (argc=1, argv=0x7fffffffd948) at ../src/network/networkd.c:130
(gdb) 

```

Reply:
OK. Thanks.